faith_data/src/commonMain/kotlin/net/thechance/mena/faith/data/repository/QuranRepositoryImpl.kt

```diff
@@ -41,4 +28,17 @@ class QuranRepositoryImpl(val ayahDao: AyahDao) : QuranRepository {
     override suspend fun saveLastAyahForTilawah(ayah: Ayah) {
         //Not yet implemented
     }
+
+    override suspend fun searchForAyahInSurah(
+        surahId: Int,
+        query: String
+    ): List<Ayah> =
+        executeLocalSafely {
+            ayahDao.searchForAyahInSurah(surahId, query).map { it.toAyah() }
```

Comment:
This approach is better for readability

```suggestion
            ayahDao.searchForAyahInSurah(surahId, query).map(AyahDto::toAyah)
```

Reply:
Sounds great.